server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ProjectResolver.kt

```diff
@@ -55,20 +56,20 @@ class ProjectResolver(
     val buildAspectResult = measured(
       "Building project with aspect"
     ) { buildProjectWithAspect(cancelChecker, workspaceContext) }
+    val aspectOutputs = measured(
+        "Reading aspect output paths"
+    ) { buildAspectResult.bepOutput.filesByOutputGroupNameTransitive(BSP_INFO_OUTPUT_GROUP) }
+    val targets = measured(
+        "Parsing aspect outputs"
+    ) { targetInfoReader.readTargetMapFromAspectOutputs(aspectOutputs).let { it } }
     val allTargetNames =
       if (buildAspectResult.isFailure)
         measured(
           "Fetching all possible target names"
-        ) { formatTargetsIfNeeded(bazelBspFallbackAspectsManager.getAllPossibleTargets(cancelChecker)) }
+        ) { formatTargetsIfNeeded(bazelBspFallbackAspectsManager.getAllPossibleTargets(cancelChecker), targets) }
```

Comment:
note: targets from`bazelBspFallbackAspectsManager.getAllPossibleTargets` are not included in `targets so they will be always mapped to `@<target>`, but i think it's not a problem

Reply:
That should work correctly as far as I can see :thinking: